src/hal/src/command/mod.rs

```diff
@@ -65,100 +64,131 @@ impl Level for Primary { }
 pub enum Secondary { }
 impl Level for Secondary { }
 
-/// Thread-safe finished command buffer for submission.
-pub struct Submit<B: Backend, C, S, L>(pub(crate) B::CommandBuffer, pub(crate) PhantomData<(C, S, L)>);
-impl<B: Backend, C, S, L> Submit<B, C, S, L> {
-    fn new(buffer: B::CommandBuffer) -> Self {
-        Submit(buffer, PhantomData)
-    }
+/// A property of a command buffer to be submitted to a queue.
+pub trait RawSubmittable<B: Backend> {
+    /// Borrow a raw command buffer from this submittable.
+    fn as_command_buffer(&self) -> &B::CommandBuffer;
 }
-unsafe impl<B: Backend, C, S, L> Send for Submit<B, C, S, L> {}
 
+/// A property of a command buffer to be submitted to a queue with specific capability.
+pub trait Submittable<B: Backend, K: Capability, L: Level>: RawSubmittable<B> {}
```

Comment:
I believe we should avoid unused bounds.

Reply:
I think these bounds are fine on a trait (as opposed to a struct). Consider them a part of documentation ;)

Reply:
It is the same problem as with struct - impl block would require those bounds even though they aren't used anywhere.

Reply:
There is a slight distinction though. A struct defines data, a trait defines logic. The constraints are more relevant to logic. Also, you typically want to implement multiple traits for a struct, hence it quickly becomes annoying. In our case, you are likely to only have at most one type to implement the trait on, so it's not nearly as annoying.